llarp/service/endpoint.cpp

```diff
@@ -734,13 +734,15 @@ namespace llarp
       if (not m_PublishIntroSet)
         return false;
 
-      auto next_pub = m_state->m_LastPublishAttempt
+      const auto lastEventAt = std::max(m_state->m_LastPublishAttempt, m_state->m_LastPublish);
+
+      auto next_pub = lastEventAt
           + (m_state->m_IntroSet.HasStaleIntros(
                  now, path::default_lifetime - path::intro_path_spread)
                  ? IntrosetPublishRetryCooldown
                  : IntrosetPublishInterval);
 
-      return now >= next_pub and m_LastIntrosetRegenAttempt + 1s <= now;
+      return now >= next_pub and lastEventAt + 1s <= now;
```

Comment:
```suggestion
      return now >= next_pub;
```
because the `lastEventAt + 1s <= now` is already implied by the value of `next_pub` (its smallest value is `lastEventAt + IntrosetPublishRetryCooldown`, and `IntrosetPublishRetryCooldown = 1s`).